File: Beecrowd/Beginner/2003 - 2176/2152 - Pepe, I Already Took the Candle!/C/2152_PepeIAlreadyTooktheCandle!.c
#include <stdio.h>

int main() {
    int testCases;
    scanf("%d", &testCases);

    while (testCases--) {
        int hour, minute, event;

        scanf("%d %d %d", &hour, &minute, &event);

        if (hour < 10) {
            printf("0%d", hour);
        } else printf("%d", hour);

        printf(":");

        if (minute < 10) {
            printf("0%d", minute);
        } else printf("%d", minute);

        if (event == 1) {
            printf(" - A porta abriu!\n");
        } else printf(" - A porta fechou!\n");
    }

    return 0;
}

File: Beecrowd/Beginner/2003 - 2176/2152 - Pepe, I Already Took the Candle!/Python/2152_PepeIAlreadyTooktheCandle!.py
testCases = int(input())

for _ in range(testCases):
    hour, minute, event = map(int, input().split())

    if hour < 10:
        print(f"0{hour}", end="")
    else:
        print(hour, end="")

    print(":", end="")

    if minute < 10:
        print(f"0{minute}", end="")
    else:
        print(minute, end="")

    if event == 1:
        print(" - A porta abriu!")
    else:
        print(" - A porta fechou!")
